include/xleaflet/test_xleaflet.hpp

```diff
@@ -0,0 +1,170 @@
+            }
+            if (py::isinstance<py::float_>(obj))
+            {
+                return obj.cast<double>();
+            }
+            if (py::isinstance<py::str>(obj))
+            {
+                return obj.cast<std::string>();
+            }
+            if (py::isinstance<py::tuple>(obj) || py::isinstance<py::list>(obj))
+            {
+                auto out = json::array();
+                for (const py::handle& value : obj)
+                {
+                    out.push_back(to_json_impl(value));
+                }
+                return out;
+            }
+            if (py::isinstance<py::dict>(obj))
+            {
+                auto out = json::object();
+                for (const py::handle& key : obj)
+                {
+                    out[py::str(key).cast<std::string>()] = to_json_impl(obj[key]);
+                }
+                return out;
+            }
+            throw std::runtime_error("to_json not implemented for this type of object: " + obj.cast<std::string>());
+        }
+    }
+
+    py::object adl_serializer<py::object>::from_json(const json& j)
+    {
+        return detail::from_json_impl(j);
+    }
+
+    void adl_serializer<py::object>::to_json(json& j, const py::object& obj)
+    {
+        j = detail::to_json_impl(obj);
+    }
+}
+
+std::vector<std::string> difference_widget(const std::set<std::string>& parsed_set, const std::set<std::string>& state_set)
+{
+    std::cout << "\nAttributes present in ipyleaflet not in xleaflet : " << "\n" << std::endl;
+
+    std::vector<std::string> difference;
+    std::set_difference(begin(parsed_set), end(parsed_set),
+                        begin(state_set), end(state_set),
+                        std::back_inserter(difference));
+    std::vector<std::string>::iterator it = difference.begin();
+    while(it != difference.end())
+    {
+        std::cout << *it << " \n";
+        it++;
+    }
+    std::cout<<std::endl;
+    return difference;
+}
+
+std::set<std::string> extract_attributes_names(const xeus::xjson& widget_representation)
+{
+    std::set<std::string> attribute_names;
+    for(const auto& el: widget_representation.items()){
+        attribute_names.insert(el.key());
+    }
+    return attribute_names;
+}
+
+namespace xlf1
+{
+    template <class WT>
+    void test_xleaflet(const std::string& ipyleaflet_widget)
+    {
+        std::cout << ipyleaflet_widget << std::endl;
+        py::object ipyleaflet = py::module::import("ipyleaflet");
+        py::object ipyleaflet_widget_state = ipyleaflet.attr(ipyleaflet_widget.c_str())().attr("get_state")();
+        xeus::xjson ipyleaflet_set = ipyleaflet_widget_state;
+        std::set<std::string> ipyleaflet_attribute_names = extract_attributes_names(ipyleaflet_set);
+
+        xeus::xjson state; xeus::buffer_sequence buffers;
+        WT().serialize_state(state, buffers);
+        std::set<std::string> xleaflet_set_item = extract_attributes_names(state);
```

Comment:
`xleaflet_set_item` -> `xleaflet_attribute_names` for homogeneity with `ipyleaflet_attribute_names`